### PR TITLE
Add serde/schemars derive to Stroke, Join and Cap.

### PR DESCRIPTION
**Pull request by ratmice**
This derives some traits for Stroke, friends.  It enables the features `schemars/smallvec`, and `serde/smallvec` so that `Dashes` within Stroke derives `JsonSchema`/`Serialize`/`Deserialize` respectively.  The impetus for this was that I was trying to add the same traits to peniko's `Style`.

### CODE REVIEW

**Review by jneem — APPROVED** [review]
I'm not actually sure if I still have write access here, but this seems pretty uncontroversial so let's try.

**Comment by ratmice** [comment]
I think that your approval worked, at least it appears to be approved in the main PR queue.
However I definitely do not have write access, so hopefully no-one is waiting for me to merge it.
I'm not in any hurry though. I just wanted to note that in case the responsibility to merge is mine.

**Pull request merged**
